models/amity.py

```diff
@@ -0,0 +1,428 @@
+        print spacer
+        emp_id = int(args["<employee_id>"])
+        for p in self.people:
+            if p.emp_id == emp_id:
+                new_person = p
+        new_room_name = args["<new_room_name>"]
+        for r in self.rooms:
+            if r.name == new_room_name:
+                new_room = r
+
+        """Check that employee ID entered exists"""
+        if emp_id not in [p.emp_id for p in self.people]:
+            print "The employee ID you have entered does not exist."
+            print "Please try again."
+            print spacer
+            return
+
+        """Check that room entered exists"""
+        if new_room_name not in [r.name for r in self.rooms]:
+            print "The room you have entered does not exist."
+            print "Please try again."
+            print spacer
+            return
+
+        """Check that room entered is vacant"""
+        if new_room_name not in [r.name for r in self.vacant_rooms]:
+            print "The room you entered, " + new_room_name + \
+                ", is not vacant at this time."
+            print "Please try again later."
+            print spacer
+            return
+
+        if new_person.job_type == "Staff":
+            """Prevent staff from being allocated living spaces"""
+            if new_room.room_type == "Living":
+                print "Staff members cannot be allocated " \
+                    "living spaces."
+                print spacer
+                return
+            else:
+                """
+                Check if staff member has already been allocated an office
+                """
+                for office in self.vacant_offices:
+                    if new_person.emp_id in \
+                            [person.emp_id for person in office.occupants]:
+                        if new_room == office:
+                            """
+                            Prevent staff member from being allocated \
+                            the same office
+                            """
+                            print new_person.name + " is already an occupant" \
+                                " of the office " + new_room.name + "."
+                            print spacer
+                            return
+                        else:
+                            """Remove staff member from current office"""
+                            office.occupants.remove(new_person)
+                """Add staff member to new office"""
+                new_room.occupants.append(new_person)
+                self.allocated_staff.append(new_person)
+                self.allocated_people.append(new_person)
+                print "You have successfully allocated " + new_person.name + \
+                    " of Employee ID " + str(new_person.emp_id) + \
+                    "\nthe following office: " + new_room.name
+
+        elif new_person.job_type == "Fellow":
+            if new_room.room_type == "Office":
```

Comment:
**Code Quality: Nitpick**

I feel like the block beneath this check here is repeated multiple times and can be compressed in a more creative way.